x-pack/plugins/index_management/public/application/components/component_templates/__jest__/client_integration/component_template_list.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { act } from 'react-dom/test-utils';
+
+import { setupEnvironment, pageHelpers, nextTick } from './helpers';
+import { ComponentTemplateListTestBed } from './helpers/component_template_list.helpers';
+import { API_BASE_PATH } from '../../../../../../common/constants';
+import { ComponentTemplateListItem } from '../../types';
+
+const { setup } = pageHelpers.componentTemplateList;
+
+jest.mock('ui/i18n', () => {
+  const I18nContext = ({ children }: any) => children;
+  return { I18nContext };
+});
+
+describe('<ComponentTemplateList />', () => {
+  const { server, httpRequestsMockHelpers } = setupEnvironment();
+  let testBed: ComponentTemplateListTestBed;
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  describe('With component templates', () => {
+    const componentTemplate1: ComponentTemplateListItem = {
+      name: 'test_component_template_1',
+      hasMappings: true,
+      hasAliases: true,
+      hasSettings: true,
+      usedBy: [],
+    };
+
+    const componentTemplate2: ComponentTemplateListItem = {
+      name: 'test_component_template_2',
+      hasMappings: true,
+      hasAliases: true,
+      hasSettings: true,
+      usedBy: ['test_index_template_1'],
+    };
+
+    const componentTemplates = [componentTemplate1, componentTemplate2];
+
+    httpRequestsMockHelpers.setLoadComponentTemplatesResponse(componentTemplates);
+
+    beforeEach(async () => {
+      testBed = await setup();
+
+      await act(async () => {
+        await nextTick(100);
```

Comment:
@sebelga what's the best way to avoid using `nextTick()` in these tests, but still account for latency with the mocked request?

Reply:
@alisonelizabeth Look at index_template_tab.test.ts in master on L37 (https://github.com/elastic/kibana/commit/c77e9bf8f88eded36028a86d49f77d1f5ebaf061#diff-29885e01c1393c73c1e16cf4a07fad91R37) I've removed all `nextTick()` calls.

You probably need to wrap the mounting (`setup()` call) inside the `act()` call.